syncer/syncer_test.go

```diff
@@ -173,80 +206,32 @@ func (s *testSyncerSuite) resetEventsGenerator(c *C) {
 	}
 }
 
-func (s *testSyncerSuite) resetBinlogSyncer() {
-	var err error
-	cfg := replication.BinlogSyncerConfig{
-		ServerID:       uint32(s.cfg.ServerID),
-		Flavor:         "mysql",
-		Host:           s.cfg.From.Host,
-		Port:           uint16(s.cfg.From.Port),
-		User:           s.cfg.From.User,
-		Password:       s.cfg.From.Password,
-		UseDecimal:     true,
-		VerifyChecksum: true,
-	}
-	if s.cfg.Timezone != "" {
-		timezone, err2 := time.LoadLocation(s.cfg.Timezone)
-		if err != nil {
-			log.L().Fatal("", zap.Error(err2))
-		}
-		cfg.TimestampStringLocation = timezone
-	}
-
-	if s.syncer != nil {
-		s.syncer.Close()
-	}
+func (s *testSyncerSuite) TearDownSuite(c *C) {}
 
-	pos, _, err := utils.GetMasterStatus(s.db, "mysql")
-	if err != nil {
-		log.L().Fatal("", zap.Error(err))
-	}
-
-	s.syncer = replication.NewBinlogSyncer(cfg)
-	s.streamer, err = s.syncer.StartSync(pos)
-	if err != nil {
-		log.L().Fatal("", zap.Error(err))
-	}
-}
-
-func (s *testSyncerSuite) TearDownSuite(c *C) {
-	s.db.Close()
-}
-
-func (s *testSyncerSuite) mockParser(db *sql.DB, mock sqlmock.Sqlmock) (*parser.Parser, error) {
-	mock.ExpectQuery("SHOW GLOBAL VARIABLES LIKE").
+func (s *testSyncerSuite) mockParser(db *sql.DB, mock *sqlmock.Sqlmock) (*parser.Parser, error) {
```

Comment:
```suggestion
func (s *testSyncerSuite) mockParser(db *sql.DB, mock sqlmock.Sqlmock) (*parser.Parser, error) {
```
seems passing sql mock by `mock sqlmock.Sqlmock` is enough and we use mock.xxx directly

Reply:
yes, I'll fix it